### PR TITLE
Replace deprecated `fmod` with `mod`

### PR DESCRIPTION
**Pull request by Clyybber**

### CODE REVIEW

**Comment by Clyybber** [comment]
No Idea why the tests are failing. It is because of this change but I don't really know why...

**Comment by skilchen** [comment]
I think because `mod` is a keyword in Nim, your simple solutiion will not work. In fact it could be a rather hard problem to get rid of `fmod` in the VM.

**Comment by GULPF** [comment]
> In fact it could be a rather hard problem to get rid of fmod in the VM.  

Even something quick & dirty like this seems to work:
```nim
proc `modd Wrapper`(a: VmArgs) {.nimcall.} =
    setResult(a, `mod`(getFloat(a, 0), getFloat(a, 1)))
registerCallback(c, "stdlib.math.mod", `modd Wrapper`)
```

IMO we should not bring back `fmod`, the `mod` keyword exists for a reason. But `fmod` must still be available in the VM for now to avoid making this a breaking change.

File: compiler/vmops.nim [hunk 1]
```diff
@@ -11,7 +11,7 @@
 #import vmdeps, vm
 from math import sqrt, ln, log10, log2, exp, round, arccos, arcsin,
   arctan, arctan2, cos, cosh, hypot, sinh, sin, tan, tanh, pow, trunc,
-  floor, ceil, fmod
+  floor, ceil, mod
```
Comment (GULPF) [inline compiler/vmops.nim]: You need to quote `mod` in this context, replace it with `` `mod` ``

Comment (GULPF) [reply compiler/vmops.nim]: And keep `fmod` as well

**Comment by skilchen** [comment]
`mod` on floats is just another name for `fmod`. 
I still don't see, why only the infix `mod` operator should be used. 
It has some advantages to make it explicit that code is dealing with floats. 

But anyway: could you @gulpf please make a working PR so that we can close this failed one? Btw. you probably don't need to invent a distinctive name for the wrapper. I think:
```nim
proc `mod Wrapper`(a: VmArgs) {.nimcall.} =
    setResult(a, `mod`(getFloat(a, 0), getFloat(a, 1)))
registerCallback(c, "stdlib.math.mod", `mod Wrapper`)
```
would also work. 

And given my limited skills, your quick&dirty solution is indeed hard.., Thank you for showing me, how it could be done!

**Comment by Clyybber** [comment]
Closing this, since IMO the workarounds aren't worth it.

**Pull request closed without merging**